source/common/res/features/enter-in-register-now/main.js

```diff
@@ -0,0 +1,61 @@
+(function poll() {
+  if (typeof ynabToolKit !== 'undefined' && ynabToolKit.actOnChangeInit === true) {
+    ynabToolKit.enterInRegisterNow = (function () {
+      let accountsController;
+      let transactionViewModel;
+
+      // this function was taken pretty much verbatim from YNAB source code
+      // they have a function called generateUpcomingTransactionNow which takes
+      // a list of entity ids to generate.
+      function enterTransactionInRegisterNow(transactionsToEnter) {
+        let entityManger = transactionViewModel.getEntityManager();
+        entityManger.openChangeSet();
+
+        let transactionIds = transactionsToEnter.map((t) => t.get('entityId'));
+
+        transactionViewModel.getYNABDatabase()
+          .generateUpcomingTransactionNow(transactionIds)
```

Comment:
Only thing I can recommend here is it probably makes sense to wrap in a `try` block, with the `catch` being to at least show an alert. Since we didn't write this function, it might go away whenever or get renamed, so it'd be awesome if the feature could fail gracefully in that case.

Other than that, awesome sleuthing!